<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Akshay-Udhane" failures="13" tests="25" name="Kafka Service Tests" time="31.772" errors="0" timestamp="2025-10-06T09:47:58 IST">
  <testcase classname="tests.KafkaServiceTest" name="testConcurrentKafkaOperations" time="0.266"/>
  <testcase classname="tests.KafkaServiceTest" name="testConvertDataValue" time="0.069">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertDataValue(KafkaServiceTest.java:124)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertDataValue -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValue" time="0.008">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValue(KafkaServiceTest.java:111)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValue -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.005">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.004">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.005">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.007">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.007">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.005">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testConvertValueWithDifferentTypes" time="0.006">
    <failure type="java.lang.AssertionError" message="Expected status 200 but got 500 expected [200] but found [500]">
      <![CDATA[java.lang.AssertionError: Expected status 200 but got 500 expected [200] but found [500]
at base.RealServiceTestBase.verifyResponseStatus(RealServiceTestBase.java:132)
at tests.KafkaServiceTest.testConvertValueWithDifferentTypes(KafkaServiceTest.java:135)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 23 stack frames]]>
    </failure>
  </testcase> <!-- testConvertValueWithDifferentTypes -->
  <testcase classname="tests.KafkaServiceTest" name="testHasChanged" time="0.008">
    <failure type="java.lang.AssertionError" message="Response is not valid JSON expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Response is not valid JSON expected [true] but found [false]
at base.RealServiceTestBase.verifyJsonResponse(RealServiceTestBase.java:160)
at tests.KafkaServiceTest.testHasChanged(KafkaServiceTest.java:64)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testHasChanged -->
  <testcase classname="tests.KafkaServiceTest" name="testHasChangedWithIdenticalData" time="0.004">
    <failure type="java.lang.AssertionError" message="Response is not valid JSON expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Response is not valid JSON expected [true] but found [false]
at base.RealServiceTestBase.verifyJsonResponse(RealServiceTestBase.java:160)
at tests.KafkaServiceTest.testHasChangedWithIdenticalData(KafkaServiceTest.java:80)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testHasChangedWithIdenticalData -->
  <testcase classname="tests.KafkaServiceTest" name="testHasChangedWithNullValues" time="0.005">
    <failure type="java.lang.AssertionError" message="Response is not valid JSON expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Response is not valid JSON expected [true] but found [false]
at base.RealServiceTestBase.verifyJsonResponse(RealServiceTestBase.java:160)
at tests.KafkaServiceTest.testHasChangedWithNullValues(KafkaServiceTest.java:99)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testHasChangedWithNullValues -->
  <testcase classname="tests.KafkaServiceTest" name="testKafkaErrorHandling" time="0.012"/>
  <testcase classname="tests.KafkaServiceTest" name="testKafkaLargeMessageHandling" time="0.01"/>
  <testcase classname="tests.KafkaServiceTest" name="testKafkaMessageProcessingPerformance" time="0.004"/>
  <testcase classname="tests.KafkaServiceTest" name="testKafkaMessageValidation" time="0.011"/>
  <testcase classname="tests.KafkaServiceTest" name="testKafkaServiceHealth" time="0.006">
    <failure type="java.lang.AssertionError" message="Kafka service should be healthy expected [true] but found [false]">
      <![CDATA[java.lang.AssertionError: Kafka service should be healthy expected [true] but found [false]
at tests.KafkaServiceTest.testKafkaServiceHealth(KafkaServiceTest.java:145)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
... Removed 21 stack frames]]>
    </failure>
  </testcase> <!-- testKafkaServiceHealth -->
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseData" time="0.005"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.004"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.006"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.005"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.007"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.004"/>
  <testcase classname="tests.KafkaServiceTest" name="testProcessBrowseDataWithDifferentTypes" time="0.005"/>
</testsuite> <!-- Kafka Service Tests -->
